### PR TITLE
Fixing mysql ALTER TABLE IF NOT EXISTS when upgrading

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

As "ALTER TABLE 'xxxx' IF NOT EXISTS" is not supported by mysql, this PR is intended to check if a column exist before trying to alters it.
image

returns :
- 0 : if it needs to be altered,
- 1 : if it's already done (eg: in a previous upgrade script),
- -1 : on errors

**Fixes** # (none)

## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [x] 19.04.x -> to be adapted (remove 19.10 script,...)
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

- Install a 19.04.0 with a mariaDB database and try to update to the latest version
- Install a 19.04.0 with a deported mysql database and try to update to the latest version
The scripts should not fail at all.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreonDB.class.php [hunk 1]
```diff
@@ -368,7 +370,7 @@ public function getProperties()
             $row = $res->fetchRow();
             $version = $row['mysql_version'];
             $info['version'] = $row['mysql_version'];
-            if ($DBRESULT = $this->query("SHOW TABLE STATUS FROM `".$this->dsn['database']."`")) {
+            if ($DBRESULT = $this->query("SHOW TABLE STATUS FROM `" . $this->dsn['database'] . "`")) {
```
Comment (lgcosta) [inline www/class/centreonDB.class.php]: `s/$DBRESULT/$DbResult/`

File: www/class/centreonDB.class.php [hunk 2]
```diff
@@ -368,7 +370,7 @@ public function getProperties()
...
                 while ($data = $DBRESULT->fetch()) {
```
Comment (lgcosta) [inline www/class/centreonDB.class.php]: `s/$DBRESULT/$DbResult/`

**Comment by sc979** [comment]
@lgcosta , changes done ;)

File: www/class/centreonDB.class.php [hunk 3]
```diff
@@ -390,4 +392,50 @@ public function getProperties()
...
+     * This method check if a column was already added in a previous upgrade script.
+     *
+     * @param string $table - the table on which we'll search the column
+     * @param string $column - the column name to be checked
+     * @return int
+     */
+    public function isColumnExist(string $table = null, string $column = null): int
+    {
+        if (!$table || !$column) {
+            $this->log->insertLog(2, 'UPGRADE PROCESS : Missing data to check if column exist');
```
Comment (ganoze) [inline www/class/centreonDB.class.php]: IMHO as you are within the global CentreonDB class, logs should be handled by the calling code, not directly by the class itself.

Comment (sc979) [reply www/class/centreonDB.class.php]: removed ;)

File: www/class/centreonDB.class.php [hunk 4]
```diff
@@ -390,4 +392,48 @@ public function getProperties()
...
+		    AND COLUMN_NAME = :columnName";
+
+        $stmt = $this->prepare($query);
+
+        try {
+            $stmt->bindValue(':dbName', $this->dsn['database'], \PDO::PARAM_STR);
+            $stmt->bindValue(':tableName', $table, \PDO::PARAM_STR);
+            $stmt->bindValue(':columnName', $column, \PDO::PARAM_STR);
+            $stmt->execute();
+            $stmt->fetch();
```
Comment (miza92) [inline www/class/centreonDB.class.php]:
```suggestion
```

Comment (miza92) [reply www/class/centreonDB.class.php]: not needed

**Comment by sc979** [comment]
Thanks @miza92 for your suggestion. I applied it ;)

**Comment by sc979** [comment]
rebased on each specific branches and added to these PR :
https://github.com/centreon/centreon/pull/7801
https://github.com/centreon/centreon/pull/7799
https://github.com/centreon/centreon/pull/7798

**Pull request closed without merging**